src/taxpasta/infrastructure/application/diamond/diamond_profile_reader.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) 2022, Moritz E. Beber, Maxime Borry, Jianhong Ou, Sofia Stamouli.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Provide a reader for diamond profiles."""
+
+import pandas as pd
+from pandera.typing import DataFrame
+
+from taxpasta.application import ProfileReader, ProfileSource
+
+from .diamond_profile import DiamondProfile
+
+
+class DiamondProfileReader(ProfileReader):
+    """Define a reader for Diamond profiles."""
+
+    LARGE_INTEGER = int(10e6)
+
+    @classmethod
+    def read(cls, profile: ProfileSource) -> DataFrame[DiamondProfile]:
+        """Read a diamond taxonomic profile from a file."""
+        nb_expected_columns = 3
+        result = pd.read_table(
+            filepath_or_buffer=profile,
+            sep="\t",
+            header=None,
+            index_col=False,
+            comment="#",
+        )
+        if len(result.columns) == nb_expected_columns:
+            result.columns = [
+                "query_id",
+                "taxonomy_id",
+                "e_value",
+            ]
```

Comment:
You could also integrate this in the `read_table` call using the `names` parameter. The only reason I didn't do this for kraken2 is that it can have either 6 or 8 columns. Pandas would then error itself if the detected columns don't match the given header columns. Also, please use the profile attributes.

```suggestion
            result.columns = [
                DiamondProfile.query_id,
                DiamondProfile.taxonomy_id,
                DiamondProfile.e_value,
            ]
```